mlfinlab/portfolio_optimization/hcaa.py

```diff
@@ -224,13 +232,130 @@ def _quasi_diagnalization(self, num_assets, curr_index):
 
         return (self._quasi_diagnalization(num_assets, left) + self._quasi_diagnalization(num_assets, right))
 
+    def _recursive_bisection(self,
+                             expected_asset_returns,
+                             asset_returns,
+                             covariance_matrix,
+                             assets,
+                             allocation_metric,
+                             optimal_num_clusters,
+                             confidence_level):
+        # pylint: disable=bad-continuation, too-many-locals
+        """
+        Recursively assign weights to the clusters - ultimately assigning weights to the individual assets.
+
+        :param expected_asset_returns: (list) a list of mean asset returns (mu)
+        :param asset_returns: (pd.DataFrame) historical asset returns
+        :param covariance_matrix: (pd.DataFrame) the covariance matrix
+        :param assets: (list) list of asset names in the portfolio
+        :param allocation_metric: (str) the metric used for calculating weight allocations
+        optimal_num_clusters: (int) optimal number of clusters for hierarchical tree clustering
+        :param confidence_level: (float) the confidence level (alpha)
+        """
+
+        num_assets = len(assets)
+        self.weights = np.ones(shape=num_assets)
+        clusters_contribution = np.ones(shape=optimal_num_clusters)
+        clusters_weights = np.ones(shape=optimal_num_clusters)
+        clusters_variance = np.ones(shape=optimal_num_clusters)
+
+        # Calculate the corresponding risk measure for the clusters
+        for cluster_index in range(optimal_num_clusters):
+            cluster_asset_indices = self.cluster_children[cluster_index]
+
+            if allocation_metric == 'minimum_variance':
```

Comment:
Lets shorted your max line length just a little bit so that the code review can fit on a single page. It makes it difficult to review if the reviewer cant see all the code without left right scroll.

Reply:
Done